pkg/client/namespaced_client_test.go

```diff
@@ -0,0 +1,597 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating the object")
+			err = getClient().Update(ctx, changedDep)
+			Expect(err).To(HaveOccurred())
+
+			deleteDeployment(ctx, changedDep, tns.Name)
+			deleteNamespace(ctx, tns)
+		})
+
+		It("should update a cluster scoped resource", func() {
+			changedCR := &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   fmt.Sprintf("clusterRole-%v", count),
+					Labels: map[string]string{"name": fmt.Sprintf("clusterRole-%v", count)},
+				},
+			}
+
+			changedCR.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "rbac.authorization.k8s.io",
+				Version: "v1",
+				Kind:    "ClusterRole",
+			})
+
+			By("Setting annotations and creating the resource")
+			changedCR.Annotations = map[string]string{"foo": "bar"}
+			changedCR, err = clientset.RbacV1().ClusterRoles().Create(ctx, changedCR, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating the deployment")
+			err = getClient().Update(ctx, changedCR)
+
+			By("validating if the cluster role was update")
+			actual, err := clientset.RbacV1().ClusterRoles().Get(ctx, changedCR.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual).NotTo(BeNil())
+			Expect(actual.Annotations["foo"]).To(Equal("bar"))
+
+			// delete cluster role resource
+			deleteClusterRole(ctx, changedCR)
+		})
+
+	})
+
+	Describe("Patch", func() {
+		var err error
+		BeforeEach(func() {
+			dep, err = clientset.AppsV1().Deployments(ns).Create(ctx, dep, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			deleteDeployment(ctx, dep, ns)
+		})
+
+		It("should successfully modify the object using Patch", func() {
+			By("Applying Patch")
+			err = getClient().Patch(ctx, dep, client.RawPatch(types.MergePatchType, generatePatch()))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating patched Deployment has new annotations")
+			actual, err := clientset.AppsV1().Deployments(ns).Get(ctx, dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual.Annotations["foo"]).To(Equal("bar"))
+			Expect(actual.GetNamespace()).To(Equal(ns))
+		})
+
+		It("should successfully modify the object using Patch when namespace is not provided", func() {
+			By("Applying Patch")
+			dep.SetNamespace("")
+			err = getClient().Patch(ctx, dep, client.RawPatch(types.MergePatchType, generatePatch()))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating patched Deployment has new annotations")
+			actual, err := clientset.AppsV1().Deployments(ns).Get(ctx, dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actual.Annotations["foo"]).To(Equal("bar"))
+			Expect(actual.GetNamespace()).To(Equal(ns))
+		})
+
+		It("should not modify the object when namespace of the object is different", func() {
+			dep.SetNamespace("non-default")
+			By("Applying Patch")
```

Comment:
I think the `By` only makes sense when you have more than one of it to split your test in logical parts